### DISCUSSION TITLE
Ability to lock objects one-by-one ?

### DISCUSSION THREAD

**Discussion by mcornillon** [ricktu288/ray-optics#147] [Ideas]
Currently there's the option to lock every objects at once.
I'd like to be able :
- either to lock all objects, but unlock one or two (the light sources for examples)
- or lock only the objects I want to lock

**Comment by mcornillon** [comment]
Maybe the new "module" function would help me in that situation, I'll will have a look at it

**Reply by ricktu288** [reply]
Yes, currently the only way to achieve this is to put the objects to be locked in a module (with absolute coordinates).

I think to implement an object-wise lock option we first need a sidebar that lists all objects, otherwise there will be no way to unlock objects. Currently there is only the JSON code editor (Settings -> Show JSON editor), and that editing the lock option (if there is one) with the code is almost as difficult as creating a module. Therefore, I think currently it is not yet suitable to implement such option.

**Reply by ricktu288** [reply]
Currently, there is another proposal requiring a GUI sidebar, which is a desmos-like module editor (see https://github.com/ricktu288/ray-optics/discussions/141#discussioncomment-9133179).

**Comment by ricktu288** [comment]
Implemented in the [sidebar branch](https://github.com/ricktu288/ray-optics/tree/sidebar). Will merge to master after the development of the sidebar is complete.

**Reply by ricktu288** [reply]
Merged into master
